src/main/java/teammates/ui/controller/StudentCourseJoinAction.java

```diff
@@ -34,8 +35,13 @@ public ActionResult execute() throws EntityDoesNotExistException {
         
         String confirmUrl = Const.ActionURIs.STUDENT_COURSE_JOIN_AUTHENTICATED 
                 + "?" + Const.ParamsNames.REGKEY + "=" + regkey 
-                + "&" + Const.ParamsNames.NEXT_URL + "=" + nextUrl;
-        data = new StudentCourseJoinConfirmationPageData(account, student, confirmUrl, Logic.getLogoutUrl(confirmUrl));
+                + "&" + Const.ParamsNames.NEXT_URL + "=" + Sanitizer.sanitizeForNextUrl(nextUrl);
+        String nextUrlType = getPageTypeOfUrl(nextUrl);
+        boolean isRedirectResult = Const.SystemParams.PAGES_ACCESSIBLE_WITHOUT_GOOGLE_LOGIN.contains(nextUrlType);
```

Comment:
It is true. Here is a case where we might want to redirect to the course join page from one of the non-```PAGES_ACCESSIBLE_WITHOUT_GOOGLE_LOGIN``` pages:

We want to promote some of our "registered features" like comments. So we allow instructors to comment on student responses. We then email the student saying that the instructor has some comments on his/her response and he/she can register with TEAMMATES to view it. We can then give them the registration link with the next url as the comments page in the email itself.